core-bundle/contao/controllers/BackendMain.php

```diff
@@ -149,7 +149,7 @@ public function run()
 		// Turbo can handle form errors.
 		$response = $this->output();
 
-		if (System::getContainer()->get('request_stack')?->getMainRequest()->attributes->has('_contao_widget_error'))
+		if (200 === $response->getStatusCode() && System::getContainer()->get('request_stack')?->getMainRequest()->attributes->has('_contao_widget_error'))
```

Comment:
```suggestion
		if (200 === $response->getStatusCode() && System::getContainer()->get('request_stack')->getMainRequest()?->attributes->has('_contao_widget_error'))
```

It is the `getMainRequest()` method that might return `null`, not the `->get('request_stack')` call.

Reply:
True - however, we do not have to check for `null` here as this is a (legacy) controller. See explanation below. Otherwise we would have to catch a `SessionNotFoundException` for `System::getContainer()->get('request_stack')->getSession()` above for example.

Reply:
Okay, but why do we check for `null` after the `->get('request_stack')` call? This smells like a workaround for unit tests, doesn't it?

Reply:
No, I think that was just a prior mistake. There are no unit tests for `BackendMain`.

Reply:
I see. I have updated my suggestion accordingly.

Reply:
I had already removed the `?` :)